### PR TITLE
Remove stubs and metakit, replacing with jars from maven central

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Testing: Check builds are green (travis and jenkins).

Travis green in this PR (obviously)
https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-build/
https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-full-repository/

All jobs green except merge-build which is yellow--needs a small javadoc link tweak (since the links to the removed components no longer work).

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-FILES-CPP-DEV-merge-push-superbuild#435](https://ci.openmicroscopy.org/job/OME-FILES-CPP-DEV-merge-push-superbuild/435/). See the [console output](https://ci.openmicroscopy.org/job/OME-FILES-CPP-DEV-merge-push-superbuild/435/consoleText) for more details.
Possible conflicts:
  - PR #2648 melissalinkert 'Add JPEG-XR support for CZI data'
    - pom.xml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-DEV-merge-push#797](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/797/). See the [console output](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/797/consoleText) for more details.
Possible conflicts:
  - PR #2648 melissalinkert 'Add JPEG-XR support for CZI data'
    - pom.xml

**Comment by hflynn** [comment]
Confirming that Travis is green and all the Bio-Formats jobs are green except BIOFORMATS-DEV-merge-build which is complaining about javadocs. Assuming there is indeed nothing else to test, this is good to merge.

**Comment by rleigh-codelibre** [comment]
Merge build is green again in https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-build/430/ after sorting out the javadocs.

**Comment by simleo** [comment]
Last commit looks fine.

**Comment by rleigh-codelibre** [comment]
https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-build/javadoc/loci/formats/services/LuraWaveService.html now has no warnings but lacks links to the stub.  Similar to the other already decoupled components.  They should be restored when we build docs for the whole collection once the decoupling is complete.

**Comment by rleigh-codelibre** [comment]
Docs fully green in https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-docs/472/

**Comment by sbesson** [comment]
@rleigh-codelibre: is there a link for a passing repository job (minimally the relevant formats subset) with the latest state of this PR included? Also what is the impact of declaring the stubs as `provided` in the bundles and should `loci_tools` be updated as well?

**Comment by rleigh-codelibre** [comment]
@sbesson Green in https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-full-repository/346/  This doesn't include the docs tweaks, but they won't affect the results.

The `provided` tweak is free of any impact outside the build.  It's the same as `compile` but is optional at runtime.  It's not transitive, which is why it needed adding here, since the dependency in formats-bsd is not used by the javadocs build. This also means it's not seen by any of the artefacts produced by bioformats_package--it's used only for building the javadocs.  So my understanding is that it's completely harmless.

**Comment by sbesson** [comment]
As these are stubs, I would also expect the impact to be minimal but maybe this is something we want to review as part of the bundle content review (added to https://trello.com/c/i1XLwx3M/29-review-bundles)

Re the repository link above, the build lasted only 20s and no test was run. I suspect the `/ome` containing the data repository was not mounted post reboot of the CI node. I just mounted it that the tests should run tomorrow.

**Comment by rleigh-codelibre** [comment]
OK. We'll see if it's all green tomorrow then.  And I'll double-check the contents of the bundles with and without the PR to be sure this is not changing anything.

(I did try to source the formats-bsd pom to get the classpath directly, but this made maven-ant-tasks return an NPE.  And it also lacks support for fetching `provided` dependencies according to the bug tracker.)

**Comment by rleigh-codelibre** [comment]
bioformats_package differences:

```
% diffstat diff
 META-INF/MANIFEST.MF                                     |    2 
 META-INF/maven/org.openmicroscopy/metakit/pom.properties |    4 
 META-INF/maven/org.openmicroscopy/metakit/pom.xml        |  481 +++++++++++++++
 ome/metakit/ColumnMap.class                              |binary
 ome/metakit/MetakitReader.class                          |binary
 5 files changed, 486 insertions(+), 1 deletion(-)
```

MANIFEST.MF: Updated commit hash only
META-INF/maven/org.openmicroscopy/metakit: Addition of pom stuff
ome/metakit/*.class: Differences in two of the class files; I diffed with the ome-metakit source and there are no source changes, so must be built slightly differently by maven in the new repo.  Probably harmless.

So looks fine--the changes are as expected, and the stubs `provided` deps are not leaking into the jar.

File: docs/sphinx/developers/components.txt [hunk 1]
```diff
@@ -255,3 +242,15 @@ All released and in-progress OME-XML schema files are contained in this
...
+
+`Stubs <https://github.com/ome/ome-stubs/tree/master>`__:
+
+Luratech LuraWave stubs and MIPAV stubs.
+
+This component provides empty classes that mirror third-party dependencies
+which are required at compile time but cannot be included in the build system
+(usually due to licensing issues).  The build succeeds since required class
+names are present with the correct method signatures; the end user is then
+expected to replace the stub .jar files at runtime.
```
Comment (mtbc) [inline docs/sphinx/developers/components.txt]: "at runtime" seems a bit late, maybe you'd do it before running?

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-full-repository/347/ should soon be green. Given that, and aside from my minor non-blocking doc quibble, seems **good to merge** to me!

**Pull request merged**